File: src/main/java/no/nav/fo/veilarbregistrering/arbeidssoker/adapter/FormidlingsgruppeGatewayConfig.java
package no.nav.fo.veilarbregistrering.arbeidssoker.adapter;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import static no.nav.sbl.util.EnvironmentUtils.getRequiredProperty;

@Configuration
public class FormidlingsgruppeGatewayConfig {
    public static final String ARENA_ORDS_API = "ARENA_ORDS_API";

    @Bean
    FormidlingsgruppeRestClient formidlingsgruppeRestClient() {
        return new FormidlingsgruppeRestClient(
                getRequiredProperty(ARENA_ORDS_API),
                () -> {return null;});
    }


}

File: src/test/kotlin/no/nav/fo/veilarbregistrering/arbeidssoker/formidlingsgruppe/adapter/FormidlingsgruppeGatewayConfig.kt
package no.nav.fo.veilarbregistrering.arbeidssoker.formidlingsgruppe.adapter

import io.mockk.mockk
import no.nav.fo.veilarbregistrering.arbeidssoker.formidlingsgruppe.adapter.FormidlingsgruppeGatewayImpl
import no.nav.fo.veilarbregistrering.arbeidssoker.formidlingsgruppe.adapter.FormidlingsgruppeRestClient
import no.nav.fo.veilarbregistrering.arbeidssoker.formidlingsgruppe.FormidlingsgruppeGateway
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class FormidlingsgruppeGatewayConfig {

    @Bean
    fun formidlingsgruppeRestClient(): FormidlingsgruppeRestClient = mockk()

    @Bean
    fun formidlingsgruppeGateway(formidlingsgruppeRestClient: FormidlingsgruppeRestClient): FormidlingsgruppeGateway {
        return FormidlingsgruppeGatewayImpl(formidlingsgruppeRestClient)
    }
}
